Anixe.Ion/Helpers/BufferWriter.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Buffers;
+
+namespace Anixe.Ion.Helpers
+{
+    internal struct BufferWriter : IDisposable
```

Comment:
pls, add method ToString()  and add the struct into Anixe.IO, we can reuse it

Reply:
BufferWriter is implemented specially for Anixe.Ion and does not handle unnecessary overhead, e.g. input validation. That's why it is internal. And incorrect usage can lead to memory leaks, e.g. if somebody will pass the struct as method parameter, JIT will create defensive copy of the struct, and both will contain the same buffer reference.

I guess you think about use case to use it for no-allocation string building.
In my opinion, better approach is to implement ValueStringBuilder used internally in .NET Core library. It uses the same idea as BufferWriter but also for short strings it uses stackallock`ated buffer.
I remember we had discussion about ValueStringBuilder in our system. The answer was to wait for it public in .NET Core library. Currently it is still an open issue in https://github.com/dotnet/runtime/issues/25587. They are waiting for language support protect of incorrect usage of this struct when accidental defensive copies are made. They need compiler feature that requires struct be passed only by reference and not by value.

I propose implement ValueStringBuilder in Anixe.IO and put a documentation comment that the struct MUST be passed by reference. I hope, the developers at ANIXE read comments ;)

@wmaryszczak  What do you think?